src/mon/MDSMonitor.cc

```diff
@@ -1443,6 +1454,22 @@ int MDSMonitor::filesystem_command(
     cmd_getval(cmdmap, "role_or_gid", who);
 
     MDSMap::mds_info_t failed_info;
+    mds_gid_t gid = gid_from_arg(fsmap, who, ss);
+    if (gid == MDS_GID_NONE) {
+      return -EINVAL;
+    }
+    if(!fsmap.gid_exists(gid, op->get_session()->get_allowed_fs_names())) {
+      // a lie; you just don't have permission to see this MDS
+      ss << "MDS named '" << who
+	 << "' does not exist, or is not up";
```

Comment:
No need "lie". You just can't get specific about the error for security reasons. Instead say:
```suggestion
	 << "' does not exist, is not up, or you lack permission to see.";
```

Reply:
Okay, fixed.